tests/Doctrine/Tests/DBAL/Functional/Ticket/GH3423Test.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Ticket;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function in_array;
+
+/**
+ * @see https://github.com/doctrine/dbal/issues/3423
+ */
+class GH3423Test extends DbalFunctionalTestCase
+{
+    /** @var bool */
+    private static $tableCreated = false;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if (! in_array($this->connection->getDatabasePlatform()->getName(), ['postgresql', 'oracle'], true)) {
```

Comment:
@morozov Yeah, I understand the fear, trust me :D I don't see another way to do it - how would I test something that the other platforms don't support? I'm not aware of a way to do that ... I think it must be sufficient that these changes don't break the existing tests as those tests should cover all the core/common functionality. And just to be extra sure I adjusted the `ConnectionTest` to let the database throw the errors, you can check it out here https://github.com/doctrine/dbal/pull/3425 .

Reply:
Here is a build on top of that improved `ConnectionTest` https://travis-ci.org/grongor/dbal/builds/479358748 - passing

Reply:
> how would I test something that the other platforms don't support? I'm not aware of a way to do that ...

Except for not being able to create a deferred constraint, how is the new test expected to behave on a platform that doesn't support deferred constraint? I.e. instead of skipping the test entirely, can we just create the deferred constraint if the platform supports it, and otherwise create a regular one?

> I think it must be sufficient that these changes don't break the existing tests as those tests should cover all the core/common functionality.

Could you identify these tests?